### ISSUE TITLE
camera up vector not respected when switching cameras

### ISSUE THREAD

**Issue by bsergean** [mattdesl/three-orbit-controls#9]
In an application that support multiple view / camera, switching camera can be done by setting the .object member variable.

```
this.controls.object = otherCamera
```

The problem is that if this camera has an up vector different from the default value, that up vector won't be respected. However, OrbitControls has support for non standard (0, 1, 0) up vectors since some code respect it to compute its internal quaternion. (cf those 2 lines of code in the "constructor").

```
var quat = new THREE.Quaternion().setFromUnitVectors( object.up, new THREE.Vector3( 0, 1, 0 ) );
var quatInverse = quat.clone().inverse();
```

A patch request follows, which make quat and quatInverse member variables and turn the previous block of code into a method that is called on update.

**Comment by mattdesl** [comment]
Ah, cool. I assume this is also pushed to ThreeJS `examples/`? I am trying to keep this mostly in sync with the ThreeJS code. :smile:

**Comment by bsergean** [comment]
Of course. Not yet submitted to three.js/example but indeed it should be
done. Last time I looked at r72 of three.js the OrbitController code had
changed quite a lot. I'm forking three.js now to apply that fix in there as
well. Sounds like another ticket to sync to the latest three.js should be
independently created.
- Benjamin

**Comment by bsergean** [comment]
Created this: https://github.com/mattdesl/three-orbit-controls/issues/11

**Comment by bsergean** [comment]
... this appear to be fixed already in the latest three.js :)

https://github.com/mrdoob/three.js/blob/master/examples/js/controls/OrbitControls.js#L203

quat and quatInverse are already updated inside of the update method. So if we update to the latest three.js code this ticket can be closed. This bug fix would only be interesting for the current release of three-orbit-controls.

**Comment by mattdesl** [comment]
I pushed this to `71.2.0`. I think we will need the same changes in the new version?

**Comment by bsergean** [comment]
We shouldn't, it looks like this was fixed already upstream but I should
test better. I actually tried the version 0.72 (I changed my code so that I
can run on three.js 0.72), and apparently with my application which does
change the up vector and changing cameras wasn't working well. I did that
in a hurry so I should try again / troubleshoot this better. npm start and
using the local test was working fine though.

In any case 71.2.0 works great for me.
- Benjamin
